subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/resolve/service/ScriptPluginUseIntegrationTest.groovy

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugin.use.resolve.service
+
+import org.gradle.api.internal.plugins.ApplyPluginBuildOperationType
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.test.fixtures.server.http.HttpServer
+import org.gradle.test.matchers.UserAgentMatcher
+import org.gradle.util.GradleVersion
+import org.junit.Rule
+
+class ScriptPluginUseIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
I think we should add some test coverage for class visibility:
- can the classes in the applying script see classes from the applied script? who else can see them? child projects? buildSrc classes? etc
- can the classes in the applied script see classes from the applying script? from buildSrc? from Gradle core? etc

Reply:
Also, when a remote script is requested from multiple projects, that the script is downloaded once rather than several times.

Reply:
There should be coverage for what happens when a plugin resolution strategy is being used and there are script plugin requests present:

- What does the resolution strategy see about the plugin request?
- What can the resolution strategy do and not do with these requests?
- What validation is applied?

Reply:
👍

Reply:
Classloading/visibility and caching of remote scripts across projects coverage added.
Resolution strategy related coverage still missing.

Reply:
Coverage for plugin resolution strategies now added.